### ISSUE TITLE
weird things happening with file manager

### ISSUE THREAD

**Issue by antonical** [authentic-theme/authentic-theme#1531]
I am logged in as the admin for the webmin/virtualmin server. I just did the following a few days ago.

New domain for client.

mavigate to virtualmin -> select new domain from list -> file manager

When in file manager I then want to upload a file to the public_html folder for this client. It uploads as root:root rather than the directory I am uploading it to. I can change the owner to the current virtual domain owner. (PITA)

When I try and extract the file(tar.gz) I get a Red error telling me: latest-en_GB.tar.gz: Provided password is incorrect

There is no password on the file. 

I am a bit confused as I setup another clients Wordpress installation just a few days ago and no issues.

Also noticed I cannot download to this directory from remote URL. When I try I get the dialog box  I can enter the URL for the file. But the remote server username and password are already filled in and not editable. If I go ahead I get a progress bar at eh bottom the says: Downloading from en-gb.wordpress.org. Please wait...

But it will sit there forever and is clearly not downloading anything.

File manager is now a PITA and is not working. I now have to workaround these issues using ssh connections etc.

Has file manger changed recently.

Cheers
Tony 

Hoping it s a config issue of some kind. Although it is OOTB config on this server.

**Comment by antonical** [comment]
Just noticed that in the public_html directory of the virtualserver. When I created a directory file manager created as root:root and not as the virtualserver owner.

**Comment by jcameron** [comment]
Did any of these issues only start to occur when you upgraded Virtualmin or Webmin?

**Comment by iliarostovtsev** [comment]
> Just noticed that in the public_html directory of the virtualserver. When I created a directory file manager created as root:root and not as the virtualserver owner.

You should check **File Manager ⇾ Preferences: Advanced options** and that _Automatically switch to user mode upon entering its home directory_ is set to _Yes_.

Afterwards, try hitting `Ctrl+Shift+R` to force reload the page and clear current cache.

**Comment by antonical** [comment]
Thank you for the feedback. Yes File Manager ⇾ Preferences: Advanced options and that Automatically switch to user mode upon entering its home directory is set to Yes.

If I copy a file from another users home directory into the current users home directory. The paste operation leaves the file as being owned by the other user. I would have expected it to save the file with the current users ownership.

Is this the expected behaviour?

Jamie. Yes both Webmin and Vmin were recently updated plus the changed we made around the hostname before it was rebooted.

**Comment by iliarostovtsev** [comment]
> If I copy a file from another users home directory into the current users home directory. The paste operation leaves the file as being owned by the other user. I would have expected it to save the file with the current users ownership.

Actually it wasn't implemented before. Fixed with this commit - https://github.com/authentic-theme/authentic-theme/commit/0c19556240181ae2e9d73da12911f5669f579454

> Thank you for the feedback. Yes File Manager ⇾ Preferences: Advanced options and that Automatically switch to user mode upon entering its home directory is set to Yes.

I think I see the bug here.

If you open browser's console and run:

```js
config_portable_module_filemanager_switch_user = true;
theme.config.save();
```

.. and reload the page afterwards, does it fix an issue?

**Comment by iliarostovtsev** [comment]
Okay, it will be fixed for upcoming theme 19.75 release.

Thanks for the heads up!
